src/core/midiMsgFilter.h

```diff
@@ -0,0 +1,84 @@
+/* -----------------------------------------------------------------------------
+ *
+ * Giada - Your Hardcore Loopmachine
+ *
+ * -----------------------------------------------------------------------------
+ *
+ * Copyright (C) 2010-2020 Giovanni A. Zuliani | Monocasual
+ *
+ * This file is part of Giada - Your Hardcore Loopmachine.
+ *
+ * Giada - Your Hardcore Loopmachine is free software: you can
+ * redistribute it and/or modify it under the terms of the GNU General
+ * Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * Giada - Your Hardcore Loopmachine is distributed in the hope that it
+ * will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Giada - Your Hardcore Loopmachine. If not, see
+ * <http://www.gnu.org/licenses/>.
+ *
+ * -------------------------------------------------------------------------- */
+
+
+#ifndef G_MIDIMSGFILTER_H
+#define G_MIDIMSGFILTER_H
+
+#include <vector>
+#include "midiMsg.h"
+
+namespace giada {
+namespace m {
+
+
+// A midiMsg filtering class
+// Allows to tell whether midiMsg is of a defined type
+//
+// TODO: Knob/slider jitter filter
+
+
+class midiMsgFilter
+{
+	private:
```

Comment:
Little coding convention: we usually list public members first, then protected, then private. Private members start with `m_` prefix, so that you don't have to put `midiMsgFilter::` in front of them.